### DISCUSSION TITLE
Move Buffer to New Window

### DISCUSSION THREAD

**Discussion by exastone** [helix-editor/helix#5592] [Ideas]
I'm Looking field interest before submitting an enhancement request for moving an already open buffer to new split window. I don't believe there is currently a streamline way of doing this after looking through the issue tracker and discussion threads, but if there is please point me to it.

The current only way of moving a file to a new window is to close the buffer in the current window, and `:vsplit` or `:hsplit` the file. (you have to first close the buffer in the existing window to prevent cycling to it in the original window with `:bn/:bp`)

This results in (possibly) a significant amount of input especially if the file is outside the current working directory.

Something such as `move_buffer_to_new_window` would be a significant time save (especially if you don't take note of the filename before closing as I often do idiotically)

Possible Implementation Issue(s):
- what should be the behaviour for such command if there is only one buffer open?
  1. do nothing (easier to implement and my prefrence)
  2. move buffer to new split window and open scratch buffer in original window

Please share your thoughts, If there is enough interest hopefully we can get this added

**Comment by CptPotato** [comment]
This doesn't address the cycling issue, but some things you might find helpful:

- For me `<space>wv` / `<space>ws` is significantly more comfortable than typing `:vsplit` / `:hsplit`
- You can use `C-v` / `C-s` to open a new split straight from the file or buffer picker

**Reply by exastone** [reply]
issue is that this is only useful if file accessible from the CWR
